docs/js/features/odata/generate-client.md

```diff
@@ -0,0 +1,115 @@
+---
+id: generator-js-sdk
+title: Generate a typed OData client for JavaScript
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Generate an OData client for JavaScript
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- JavaScript 
+- TypeScript
+- generate
+author: Charles Dubois
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Generate a typed OData client with the OData Generator
+
+The generator allows you to generate a custom OData client library for any service. You can then easily access these services via the client libraries.
+
+There are two way to use the generator:
+- Via the command line interface (CLI)
+- By instantiating and invoking it at runtime
+
+For any of them you need to have an `EDMX` file holding the service metadata.
+
+## Installation
+
+Run this command in your project's terminal to install the generator globally.
+```sh
+npm install -g @sap-cloud-sdk/generator
+```
+
+:::note
+The generator is included in the SAP cloud-sdk-cli, if you have it installed you don't need to install the generator.
+:::
+
+## Usage
+
+### Using the CLI
+
+The generator is primarily meant to be used on the command line:
+
+```sh
+generate-odata-client --inputDir path/to/your/service-specification(s) --outputDir path/where/the/modules/are/stored
+```
+
+Run `generate-odata-client --help` for further options.
+
+### Invoke the generator programmatically
+
+You can also use the generator programmatically. You will have to provide the options anyways.
```

Comment:
Why `anyways`?

I would recommend to write the code below as one code snippet with some comments and describe it in a text below. That ways it becomes clearer what the actual code is and people can copy it to try.